spec/system/comments/user_fills_out_comment_spec.rb

```diff
@@ -46,4 +46,71 @@
     find(:xpath, "//div[contains(@class, 'reply-actions')]/input[@name='commit']").click
     expect(page).to have_text(raw_comment)
   end
+
+  # This is basically a black box test for
+  # ./app/javascripts/packs/validateFileInputs.js
+  # which is logic to validate file size and format when uploading via a form.
+  it "User attaches a valid image" do
+    visit article.path.to_s
+
+    wait_for_javascript
+
+    attach_file(
+      "image-upload-main",
+      Rails.root.join("app/assets/images/sloan.png"),
+      visible: false,
+    )
+
+    expect(page).to have_no_css("div.file-upload-error")
+  end
+
+  # This is basically a black box test for
+  # ./app/javascripts/packs/validateFileInputs.js
+  # which is logic to validate file size and format when uploading via a form.
+  it "User attaches a large image" do
+    visit article.path.to_s
+
+    wait_for_javascript
```

Comment:
In the interest of not slowing down the test suite, are we sure we need this? Capybara is generally quite good at waiting for JS by itself. Not saying we don't, just want to make sure, as I've been working on making the test suite faster lately.

Reply:
Good call - removed!